mscore/scoretreeview.cpp

```diff
@@ -0,0 +1,91 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2
+//  as published by the Free Software Foundation and appearing in
+//  the file LICENCE.GPL
+//=============================================================================
+
+#include "scoretreeview.h"
+
+namespace Ms {
+
+//---------------------------------------------------------
+//   ScoreTreeView
+//---------------------------------------------------------
+
+ScoreTreeView::ScoreTreeView(QWidget* parent)
+    : QTreeView(parent)
+{
+}
+
+//---------------------------------------------------------
+//   ~ScoreTreeView
+//---------------------------------------------------------
+
+ScoreTreeView::~ScoreTreeView()
+{
+}
+
+//---------------------------------------------------------
+//   setModel
+//---------------------------------------------------------
+
+void ScoreTreeView::setModel(QAbstractItemModel* model)
+{
+    ScoreItemModel* scoreItemModel = dynamic_cast<ScoreItemModel*>(model);
+    Q_ASSERT(scoreItemModel);
+    QTreeView::setModel(scoreItemModel);
+}
+
+//---------------------------------------------------------
+//   model
+//---------------------------------------------------------
+
+ScoreItemModel* ScoreTreeView::scoreItemModel() const {
+    return static_cast<ScoreItemModel*>(QTreeView::model());
+}
+
+//---------------------------------------------------------
+//   currentChanged
+//---------------------------------------------------------
+
+void ScoreTreeView::currentChanged(const QModelIndex& current, const QModelIndex& previous)
+{
+    selectElementAtIndex(current);
+    QTreeView::currentChanged(current, previous);
+}
+
+//---------------------------------------------------------
+//   selectElementAtIndex
+//---------------------------------------------------------
+
+void ScoreTreeView::selectElementAtIndex(const QModelIndex& index)
+{
+    ScoreElement* s = (ScoreElement*) index.internalPointer();
```

Comment:
You should never use C-style casts. In this case `static_cast` is appropriate, or you could do `treeItemModel()->scoreElementFromIndex(index)` (but that's probably overkill).